### PR TITLE
Rename mgmt_hdr to smp_hdr and make it internal

### PR DESCRIPTION
**Pull request by de-nordic**
Only SMP protocol layer, with exception to SMP reassembly which is part of transport layer, is supposed to be processing packets, so it has to know how does the header looks like.
The mgmt_hdr is actually SMP protocol header, as described here https://docs.zephyrproject.org/latest/services/device_mgmt/smp_protocol.html#frame-the-envelope, and the name is a little bit misleading as mgmt layer of MCUmgr does not process it.

This PR:
 1) ~moves `mgmt_hton_hdr` and `mgmt_ntoh_hdr` functions to smp,c and renames them `smp_hdr_hton` and `smp_hdr_ntoh` respectively~; removes `mgmt_hton_hdr` and `mgmt_ntoh_hdr`.
 2) moves `mgmt_hdr` definition to smp_internal.h, making it internal
 3) redefines `mgmt_hdr` as `smp_hdr`
 4) fixes tests.
 5) unifies function definition style in smp.c

~~Depends on https://github.com/zephyrproject-rtos/zephyr/pull/50602~~

### CODE REVIEW

File: subsys/mgmt/mcumgr/lib/smp/src/smp.c [hunk 1]
```diff
@@ -1,23 +1,59 @@
...
 #include "mgmt/mgmt.h"
 #include <zcbor_common.h>
+#include <zcbor_decode.h>
 #include <zcbor_encode.h>
 #include "smp/smp.h"
 #include "../../../smp_internal.h"
 
+static void
+cbor_nb_reader_init(struct cbor_nb_reader *cnr,
+		    struct net_buf *nb)
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/smp/src/smp.c:25]: Fix styling here, no need to keep the old weird library styling

File: subsys/mgmt/mcumgr/lib/smp/src/smp.c [hunk 2]
```diff
@@ -1,23 +1,59 @@
...
+	void *new_ptr = net_buf_pull(nb, sizeof(struct smp_hdr));
+
+	cnr->nb = nb;
+	zcbor_new_decode_state(cnr->zs, ARRAY_SIZE(cnr->zs), new_ptr,
+			       cnr->nb->len, 1);
+}
+
+static void
+cbor_nb_writer_init(struct cbor_nb_writer *cnw, struct net_buf *nb)
+{
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/smp/src/smp.c:37]: And here, etc.

File: subsys/mgmt/mcumgr/smp.c [hunk 3]
```diff
@@ -8,11 +8,11 @@
 #include <zephyr/kernel.h>
 #include <zephyr/device.h>
 #include <zephyr/net/buf.h>
-#include <zephyr/mgmt/mcumgr/buf.h>
-#include "mgmt/mgmt.h"
-#include "smp/smp.h"
+#include <zcbor_common.h>
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/smp.c:11]: Don't think zcbor header is needed here, can't see any usage of zcbor structs in this file

**Review by nordicjm — CHANGES_REQUESTED** [review]

File: subsys/mgmt/mcumgr/smp.c (hunk 3 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/smp.c:11]: It is no longer there.

File: subsys/mgmt/mcumgr/lib/smp/src/smp.c [hunk 4]
```diff
@@ -18,33 +20,41 @@
...
 			       net_buf_tailroom(nb), 0);
 }
 
+static void smp_hdr_ntoh(struct smp_hdr *hdr)
+{
+	hdr->nh_len = sys_be16_to_cpu(hdr->nh_len);
+	hdr->nh_group = sys_be16_to_cpu(hdr->nh_group);
+}
+
+static void smp_hdr_hton(struct smp_hdr *hdr)
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/smp/src/smp.c:48]: This is used for tests and cannot be made static: https://github.com/zephyrproject-rtos/zephyr/pull/44124/files#diff-13980dedd3fed9aaa5b48c06295ce8ecc9c7fa3968fd4322d6212435b3ca0f9aR24

Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/smp/src/smp.c:48]: This is internal, auxiliary,  function that should have never been exported. The test implements SMP client and can spare some lines (2) to implement the feature.

File: subsys/mgmt/mcumgr/lib/smp/src/smp.c [hunk 5]
```diff
@@ -53,22 +63,20 @@ smp_rsp_op(uint8_t req_op)
...
 
-static void
-smp_make_rsp_hdr(const struct mgmt_hdr *req_hdr, struct mgmt_hdr *rsp_hdr, size_t len)
+static void smp_make_rsp_hdr(const struct smp_hdr *req_hdr, struct smp_hdr *rsp_hdr, size_t len)
 {
-	*rsp_hdr = (struct mgmt_hdr) {
+	*rsp_hdr = (struct smp_hdr) {
 		.nh_len = len,
 		.nh_flags = 0,
 		.nh_op = smp_rsp_op(req_hdr->nh_op),
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/smp/src/smp.c:71]: In this case, does it make more sense to get rid of `smp_hdr_hton()` entirely and use the `sys_*()` functions instead?

File: subsys/mgmt/mcumgr/lib/smp/src/smp.c [hunk 6]
```diff
@@ -305,7 +307,7 @@ smp_process_request_packet(struct smp_streamer *streamer, void *vreq)
 		} else {
 			valid_hdr = true;
 		}
-		mgmt_ntoh_hdr(&req_hdr);
```
Comment (nordicjm) [inline subsys/mgmt/mcumgr/lib/smp/src/smp.c:308]: Likewise here, can just manually flip len and group if needed

File: subsys/mgmt/mcumgr/lib/smp/src/smp.c (hunk 5 above)
Comment (de-nordic) [reply subsys/mgmt/mcumgr/lib/smp/src/smp.c]: Yes.

**Review by nordicjm — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**